File: src/main/scala/org/xbery/artbeams/orders/repository/OrderItemFilter.scala
package org.xbery.artbeams.orders.repository

import org.xbery.artbeams.common.assets.repository.AssetFilter

/**
  * @author Radek Beran
  */
case class OrderItemFilter(
  override val id: Option[String],
  override val ids: Option[Seq[String]],
  override val createdBy: Option[String],
  orderId: Option[String],
  productId: Option[String]
) extends AssetFilter

object OrderItemFilter {
  lazy val Empty = OrderItemFilter(None, None, None, None, None)
}

File: src/main/kotlin/org/xbery/artbeams/orders/repository/OrderItemFilter.kt
package org.xbery.artbeams.orders.repository

import org.xbery.artbeams.common.assets.repository.AssetFilter

/**
 * @author Radek Beran
 */
data class OrderItemFilter(
    override val id: String?,
    override val ids: List<String>?,
    override val createdBy: String?,
    val orderId: String?,
    val productId: String?
) : AssetFilter {
    companion object {
        val Empty = OrderItemFilter(null, null, null, null, null)
    }
}
